cmd/wksctl/apply/apply.go

```diff
@@ -66,16 +66,26 @@ type Applier struct {
 }
 
 func (a *Applier) Apply() error {
-	var closer func()
-	var err error
-	cpath := filepath.Join(a.Params.gitPath, a.Params.clusterManifestPath)
-	mpath := filepath.Join(a.Params.gitPath, a.Params.machinesManifestPath)
-
-	cpath, mpath, closer, err = manifests.Get(cpath, mpath, a.Params.gitURL, a.Params.gitBranch, a.Params.gitDeployKeyPath, a.Params.gitPath)
-	if err != nil {
-		return err
+	var cpath, mpath string
+
+	if a.Params.gitURL == "" {
+		// Cluster and Manifests come from the local filesystem.
+		cpath, mpath = a.Params.clusterManifestPath, a.Params.machinesManifestPath
+	} else {
+		// Cluster and Machine manifests come from a Git repo that we'll clone for the duration of this command.
+		repo, err := manifests.CloneClusterAPIRepo(a.Params.gitURL, a.Params.gitBranch, a.Params.gitDeployKeyPath, a.Params.gitPath)
+		if err != nil {
+			return errors.Wrap(err, "manifests.Get")
+		}
+		defer repo.Close()
+
+		if cpath, err = repo.ClusterManifestPath(); err != nil {
+			return errors.Wrap(err, "ClusterManifestPath")
+		}
+		if mpath, err = repo.MachinesManifestPath(); err != nil {
+			return errors.Wrap(err, "MachinesManifestPath")
+		}
 	}
-	defer closer()
```

Comment:
just verifying, but this ideally should have been in the previous commit, right?  I ask to make sure there aren't other places where we need to remove `defer closer()` before any `initiateCluster` calls that we yet haven't.

Reply:
Not sure what you mean. 4177a0b94140cc9ee93fdd3d6cd3fe2ecf7318a0 was supposed to be an atomic change that moves the `defer` statement from the callee (`initiateCluster`) to the caller. There's an identical change in be6530490145a2acc43a8171e8f44a6c84f0be7d.

Reply:
ahh. ok.  just wanted to be totally sure :) thanks!